thunder/core/prims.py

```diff
@@ -480,7 +480,7 @@ def _check_tensor_shape_and_metadata_meta(
 ) -> None:
     # Validates types
     baseutils.check_type(t, TensorProxy)
-    baseutils.check_valid_shape(shape)
+    # we don't need to validate shape here, because we use `-1` for dynamic shape
```

Comment:
Can't we still validate the numbers in the shape, and that the shape is composed of integers and number proxies?

Reply:
Also: what about changing the annotation for this function?

Reply:
I'll merge this conversation with the below one. Since I'm only allowing numbers, but replacing NumberProxy with `-1`.